File: cpp/153_FindMinimuminRotatedSortedArray.cpp
class Solution {
public:
    int findMin(vector<int>& nums) {
        int l = 0, h = nums.size()-1;
        while(nums[l] > nums[h]) {
            int m = (l+h)/2;
            if(nums[l] <= nums[m]) l = m+1;
            else h = m;
        }
        return nums[l];
    }
};

File: js/153_FindMinimuminRotatedSortedArray.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var findMin = function(nums) {
    var l = 0, h = nums.length-1;
    while(nums[l] > nums[h]){
        var m = Math.floor( (l+h)/2 );
        if(nums[l] <= nums[m]) l = m+1;
        else h = m;
    }
    return nums[l];
};